### PR TITLE
Added back button navigation

### PR DESCRIPTION
**Pull request by KaviHarjani**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -6,59 +6,61 @@ void main() {
...
-      navigationDelegate: (NavigationRequest request) {
-        if (request.url.startsWith(_screenerHomeUrl)) {
-          return NavigationDecision.navigate;
-        } else {
-          _launchURL(request.url);
-          return NavigationDecision.prevent;
-        }
-      },
-    );
+    return MaterialApp(
```
Comment (manujagobind) [inline lib/main.dart:17]: Let's maintain smaller widgets as before?

Comment (KaviHarjani) [reply lib/main.dart:17]: Hey @manujagobind  i tried doing this same thing with smaller widget tree, maybe I am not well versed with flutter but it took me 2-3 days and still I wasn't able to figure out, 
So I switched back it and got it done, and it was easier that way for me

Comment (manujagobind) [reply lib/main.dart:17]: Ok, no worries. I'll give it a try on this branch.

Comment (KaviHarjani) [reply lib/main.dart:17]: Hey @manujagobind don't worry about it, I'll do it all at the end,
It's easy thing to do, I've to just extract the widgets with 'Ctrl + .'
The thing is I find it hard to work that way, so let's keep it simple for now?

Comment (manujagobind) [reply lib/main.dart:17]: I can help you understand the split.

Comment (KaviHarjani) [reply lib/main.dart:17]: sure, but not like i don't get the split, I can read it and get what it does, but it's just hard working with it
for example, try adding the back navigation with the split

I can do the split with "Ctrl +."
Maybe do that towards the end?

Comment (ashank96) [reply lib/main.dart:17]: Widgets split is a standard idea. The longer the widgets, the higher chances that widget tree takes time to render, if there are many states across the unit widgets. 
Extracting the widgets into smaller ones, delegates state handling to the atomic widgets and not to the whole widget tree, hence optimised rendering as well, apart from just code readability.

Comment (KaviHarjani) [reply lib/main.dart:17]: Hey @ashank96 can you push the changes?

**Comment by manujagobind** [comment]
Closing this experiment branch for now.

**Pull request closed without merging**